pyterrier/rewrite.py

```diff
@@ -309,3 +309,65 @@ def transform(self, queries_and_docs):
         self.qe.fbDocs = self.fb_docs
         return super().transform(queries_and_docs)
 
+def linear(weightCurrent : float, weightPrevious : float, format="terrierql", **kwargs) -> TransformerBase:
+    """
+    Applied to make a linear combination of the current and previous query formulation. The implementation
+    is tied to the underlying query language used by the retrieval/re-ranker transformers. Two of Terrier's
+    query language formats are supported by the `format` kwarg, namely `"terrierql"` and `"matchoptql"`. 
+    Their exact respective formats are `detailed in the Terrier documentation <https://github.com/terrier-org/terrier-core/blob/5.x/doc/querylanguage.md>`_.
+
+    Args:
+        weightCurrent(float): weight to apply to the current query formulation.
+        weightPrevious(float): weight to apply to the previous query formulation.
+        format(str): which query language to use to rewrite the queries, one of "terrierql" or "matchopql".
+
+    Example::
+
+        pipeTQL = pt.apply.query(lambda row: "az") >> pt.rewrite.linear(0.75, 0.25, format="terrierql")
+        pipeMQL = pt.apply.query(lambda row: "az") >> pt.rewrite.linear(0.75, 0.25, format="matchopql")
+        pipeT.search("a")
+        pipeM.search("a")
+
+    Example outputs of `pipeTQL` and `pipeMQL` corresponding to the query "a" above:
+
+    - Terrier QL output: `"(az)^0.750000 (a)^0.250000"`
+    - MatchOp QL output: `"#combine:0:0.750000:1:0.250000(#combine(az) #combine(a))"`
+
+    """
+    return _LinearRewriteMix([weightCurrent, weightPrevious], format, **kwargs)
+
+class _LinearRewriteMix(TransformerBase):
+
+    def __init__(self, weights : List[float], format : str = 'terrierql', **kwargs):
+        super().__init__(**kwargs)
+        self.weights = weights
+        self.format = format
+        if format not in ["terrierql", "matchopql"]:
+            raise ValueError("Format must be one of 'terrierql', 'matchopql'")
+
+    def _terrierql(self, row):
+        return "(%s)^%f (%s)^%f" % (
```

Comment:
should we be concerned about escaping the query text, is it safe to assume it's already in the target query language?

Reply:
I think its safe to assume its already in the correct query language.

Reply:
alright, then lgtm